### PR TITLE
Update CONF layer

### PR DESCRIPTION
**Pull request by nemo794**
This PR updates the CONF layer to use the following data classification values:

```
WATER_NOT_WATER_CLEAR = 0
WATER_UNCOLLAPSED_HIGH_CONF_CLEAR = 1
WATER_UNCOLLAPSED_MODERATE_CONF_CLEAR = 2
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_CONSERVATIVE_CLEAR = 3
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_AGGRESSIVE_CLEAR = 4

WATER_NOT_WATER_CLOUD = 10
WATER_UNCOLLAPSED_HIGH_CONF_CLOUD = 11
WATER_UNCOLLAPSED_MODERATE_CONF_CLOUD = 12
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_CONSERVATIVE_CLOUD = 13
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_AGGRESSIVE_CLOUD = 14

WATER_NOT_WATER_SNOW = 20
WATER_UNCOLLAPSED_HIGH_CONF_SNOW = 21
WATER_UNCOLLAPSED_MODERATE_CONF_SNOW = 22
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_CONSERVATIVE_SNOW = 23
WATER_UNCOLLAPSED_PARTIAL_SURFACE_WATER_AGGRESSIVE_SNOW = 24
```

It also updates the names of several global constants throughout the code.

Attached are two sample output directories, excluding the *DEM.tif files because they are too large. The Tahoe files illustrate snow, and the Taiwan directory displays the ocean masking.

[tahoe_output_dir.zip](https://github.com/gshiroma/PROTEUS/files/10365469/tahoe_output_dir.zip)
[taiwan_output_dir.zip](https://github.com/gshiroma/PROTEUS/files/10365471/taiwan_output_dir.zip)

### CODE REVIEW

File: src/proteus/dswx_hls.py
Comment (gshiroma) [inline src/proteus/dswx_hls.py]: Classifications or classification? Maybe we can just call it `Confidence layer`

File: src/proteus/dswx_hls.py
Comment (gshiroma) [inline src/proteus/dswx_hls.py]:
```suggestion
    cloud_idx = ((cloud_layer == 1) | 
                 (cloud_layer == 3) |
                 (cloud_layer == 4) |
                 (cloud_layer == 5) |
                 (cloud_layer == 6) |
                 (cloud_layer == 7))
```

File: src/proteus/dswx_hls.py
Comment (gshiroma) [inline src/proteus/dswx_hls.py]:
```suggestion
    The cloud classification in the CONF layer represents
    the ensemble of cloud, cloud shadow, or adjacent-to-cloud
    classification, and it has precedence over snow.
    A pixel is only marked as snow if it's not 
    cloud/cloud-shadow/adjacent
    masked. (e.g., a pixel with a value of 6 in the CLOUD layer
    would be marked as cloud in the CONF layer.)
```

**Review by gshiroma — APPROVED** [review]
LGTM! Thank you, @nemo794 ! I like how you updated the creation of the CONF layer and its color table. I also appreciate your attention to commenting the code thoroughly.

File: src/proteus/dswx_hls.py
Comment (nemo794) [reply src/proteus/dswx_hls.py]: I kinda like calling it "classification" (singular), to help people remember that this values do not reflect percentages. I'll update to "classification" now, but if you feel strongly about changing it in your PR after this, "Confidence layer" makes sense too.

**Comment by nemo794** [comment]
Thank you for your review, @gshiroma !

As noted in one of my responses, if you'd like to update the description for the CONF layer again, please go for it before merging your PR into nasa/PROTEUS.

Otherwise, it looks like only folks with write access to this repo (gshiroma) are able to merge this PR. Please merge when you're ready. Thank you again!

**Comment by gshiroma** [comment]
> Thank you for your review, @gshiroma !
> 
> As noted in one of my responses, if you'd like to update the description for the CONF layer again, please go for it before merging your PR into nasa/PROTEUS.
> 
> Otherwise, it looks like only folks with write access to this repo (gshiroma) are able to merge this PR. Please merge when you're ready. Thank you again!

Thank you, @nemo794 ! I'm going to merge it now

**Pull request merged**
